client/views/cards/events.js

```diff
@@ -207,6 +210,39 @@ Template.cardLabelsPop.events({
         };
         Cards.update(cardId, query);
         event.preventDefault();
+    },
+    'click .js-edit-label': function(event, tpl) {
+        Utils.Pop.open('editLabelPop', 'Change Label', $('.openPop').get(0), _.extend({ boardId: tpl.data.boardId }, this));
+        event.preventDefault();
+    }
+});
+
+Template.editLabelPop.events({
+    'submit .edit-label': function(event, tpl) {
+        var name = tpl.find('#labelName').value,
+            getLabel = Utils.getLabelIndex(this.boardId, this._id),
+            selectLabel = Blaze.getData(tpl.$('.js-palette-select:not(.hide)').get(0)),
+            $set = {};
+
+        // set label index 
+        $set[getLabel.key('name')] = name;
+
+        // color current not currentcolor
+        $set[getLabel.key('color')] = selectLabel.color;
+        
+        // update
+        Boards.update(this.boardId, { $set: $set });   
```

Comment:
Ok, so let's keep it like this.

That does not makes sense to me, I asked the question on the mailing list, https://groups.google.com/d/msg/meteor-talk/qUa3clv4g5E/R6TziALdUbwJ